src/sentry/eventstore/snuba/backend.py

```diff
@@ -56,10 +59,74 @@ def get_event_by_id(self, project_id, event_id, additional_columns=None):
 
         return SnubaEvent.get_event(project_id, event_id, snuba_cols=cols)
 
+    def get_next_event_id(self, event, conditions=None, filter_keys=None):
+        """
+        Returns (project_id, event_id) of a next event given a current event
+        and any filters/conditions. Returns None if no next event is found.
+        """
+
+        if not event:
+            return None
+
+        time_condition = [
+            ['timestamp', '>=', event.timestamp],
+            [['timestamp', '>', event.timestamp], ['event_id', '>', event.event_id]]
```

Comment:
Damn.... this is probably not going to work for multi project

Reply:
On second thoughts this is probably fine, we can keep sorting by event ID in the case of same timestamps with multi project

Reply:
Why wouldn't this work for multiproject queries? Outside of timestamp and eventid we don't really have much else to sort on.

Reply:
Yeah, i was a bit confused earlier. I think it's all good.